sycl/include/CL/sycl/detail/kernel_desc.hpp

```diff
@@ -33,8 +33,9 @@ enum class kernel_param_kind_t {
   kind_accessor = 0,
   kind_std_layout = 1, // standard layout object parameters
   kind_sampler = 2,
-  kind_pointer = 3,
-  kind_specialization_constants_buffer = 4,
+  kind_stream = 3,
+  kind_pointer = 4,
+  kind_specialization_constants_buffer = 5,
```

Comment:
Yes. It also aligns with the enum kernel_param_kind_t  declared in Sema.h.

Reply:
Two things: `kernel_param_kind_t::kind_stream` seems to be `5`, not `3` (so it's not aligned, unless I'm missing something), and if "yes" means this can impact ABI, we should probably add it to the end of the list here like we did in Sema.h.